src/main/flight/flight_health.c

```diff
@@ -0,0 +1,131 @@
+
+#include "config/feature.h"
+#include "config/config.h"
+
+#include "flight_health.h"
+
+#define THRUST_IMBALANCE_GYRO_RATE_MAX 200
+#define THRUST_IMBALANCE_THROTTLE_MIN 10
+#define THRUST_IMBALANCE_TEMP_DISABLE_US 500000 // 0.5 sec
+
+// Becomes `true` when sum of absolute I terms for all axes exceeds the threshold.
+// This indicates a high likelihood of issues with props or motors, signaling the need for mechanical inspection.
+static bool thrustImbalanceDetected = false;
+
+bool isThrustImbalanceDetected(void)
+{
+#ifdef USE_THRUST_IMBALANCE_DETECTION
+    return thrustImbalanceDetected;
+#else
+    return false;
+#endif    
+}
+
+#ifdef USE_THRUST_IMBALANCE_DETECTION
+void thrustImbalanceDetectionProcess(timeUs_t currentTimeUs)
+{
+    static timeUs_t thrustImbalanceTriggerUs = 0;
+    static timeUs_t thrustImbalanceUntriggerUs = 0;
+    static timeUs_t thrustImbalanceDisabledUntilUs = 0;
+
+    if (ARMING_FLAG(ARMED)
+        && !isFixedWing()
+        && currentPidProfile->thrust_imbalance_threshold > 0
+        && !isFlipOverAfterCrashActive()) {
+
+        // Convert threshold to the working domain.
+        const float threshold = currentPidProfile->thrust_imbalance_threshold * 1e5f;
+        // Convert tenths to us.
+        const timeUs_t triggerDelayUs = currentPidProfile->thrust_imbalance_trigger_delay * 100000;
+        // Convert tenths to us.
+        const timeUs_t untriggerDelayUs = currentPidProfile->thrust_imbalance_untrigger_delay * 100000;
+
+        if (calculateThrottlePercentAbs() < THRUST_IMBALANCE_THROTTLE_MIN
+            || gyroAbsRateDps(FD_ROLL) > THRUST_IMBALANCE_GYRO_RATE_MAX
+            || gyroAbsRateDps(FD_PITCH) > THRUST_IMBALANCE_GYRO_RATE_MAX
+            || gyroAbsRateDps(FD_YAW) > THRUST_IMBALANCE_GYRO_RATE_MAX) {
+            thrustImbalanceDisabledUntilUs = currentTimeUs + THRUST_IMBALANCE_TEMP_DISABLE_US;
+        }
+
+        const bool detectionEnabled = cmpTimeUs(currentTimeUs, thrustImbalanceDisabledUntilUs) >= 0;
+        if (detectionEnabled) {
+            thrustImbalanceDisabledUntilUs = 0;
+        }
+
+        // Multiplication of I terms gives us a measure of total thrust imbalance in all of the axes.
+        // A bad or loose prop will cause coupled I term buildup in roll, pitch, and yaw at the same time.
+        const float iMult = fabsf(pidData[FD_ROLL].I * pidData[FD_PITCH].I * pidData[FD_YAW].I);
+
+        if (detectionEnabled && iMult >= threshold) {
+            if (thrustImbalanceTriggerUs == 0) {
+                thrustImbalanceTriggerUs = currentTimeUs + triggerDelayUs;
+            }
+            if (cmpTimeUs(currentTimeUs, thrustImbalanceTriggerUs) >= 0) {
+                thrustImbalanceUntriggerUs = currentTimeUs + untriggerDelayUs;
+                thrustImbalanceDetected = true;
+            }
+        } else {
+            thrustImbalanceTriggerUs = 0;
+            if (thrustImbalanceUntriggerUs != 0 && cmpTimeUs(currentTimeUs, thrustImbalanceUntriggerUs) >= 0) {
```

Comment:
first 'balanced' sample will thrustImbalanceTriggerUs = 0, stoping thrustImbalanceUntriggerUs updating.
second 'balance' sample (after thrustImbalanceUntriggerUs delay) will reset thrustImbalanceDetected

(untriggerDelayUs > triggerDelayUs will probably mask this problem)